.github/workflows/build-and-test.yml

```diff
@@ -46,43 +46,61 @@ jobs:
           yarn build
 
       # Execute the tests.
-      - name: Test
+      - name: Run Unit Tests
         run: yarn test
         env:
           # The test result file name can be controlled using the following environment variable.
           JEST_JUNIT_OUTPUT_NAME: unit-test-results-${{ runner.os }}.xml
 
-      # Upload Test Results (The different files for the OSes will end up in the same artifact).
-      - name: Upload Test Results
+      # Upload Unit Test Results (The different files for the OSes will end up in the same artifact).
+      - name: Upload Unit Test Results
         if: always()
         uses: actions/upload-artifact@v3
         with:
           name: unit-test-tesults
           # Include the unit-test-results folders (which is in the root of the workspace).
           path: unit-test-results
+          retention-days: 30
+
+      # Run PlayWright tests, only on Linux container.
+      - name: Install Playwright Browsers
+        if: runner.os == 'Linux'
+        run: yarn --cwd ./e2e-tests/ playwright install --with-deps
+      - name: Run Playwright tests
+        if: runner.os == 'Linux'
+        uses: coactions/setup-xvfb@v1
+        with:
+          run: yarn --cwd ./e2e-tests/ test
+      - name: Upload PlayWrite test report
+        if: always() && runner.os == 'Linux'
+        uses: actions/upload-artifact@v3
+        with:
+          name: allure-results
+          path: e2e-tests/allure-results/
+          retention-days: 30
 
       # Run lint only on Linux (since it only makes sense to run it once, and linux is the fastest).
       - name: Lint
-        if: runner.os == 'Linux'
+        if: always() && runner.os == 'Linux'
```

Comment:
Could you elaborate on what is the difference between this line and just `runner.os == 'Linux'`? It seems to me that sometimes we use the combination and sometimes we do not.

Reply:
@martin-fleck-at Like discusses, the always() is needed so it also runs when any previous step has failed.